datacube/model/__init__.py

```diff
@@ -718,8 +718,11 @@ def grid_range(lower, upper, step):
         >>> list(GridSpec.grid_range(1.0, 4.0, 3.0))
         [0, 1]
         """
-        assert step > 0.0
-        return range(int(math.floor(lower / step)), int(math.ceil(upper / step)))
+        assert abs(step) > 0.0
```

Comment:
That's much smarter than what I was doing. New commit adds that example in doctest format and I switched to the implementation you suggested.